tests/integrations/basic-fail/Cargo.stdout

```diff
@@ -755,6 +756,57 @@ full stderr:
 full stdout:
 
 
+
+FAILED TEST: tests/actual_tests_bless/wrong_diagnostic_code.rs
+command: "rustc" "--error-format=json" "--extern" "basic_fail=$DIR/$DIR/../../../target/$TMP/$TRIPLE/debug/libbasic_fail.rlib" "--extern" "basic_fail=$DIR/$DIR/../../../target/$TMP/$TRIPLE/debug/libbasic_fail-$HASH.rmeta" "-L" "$DIR/$DIR/../../../target/$TMP/$TRIPLE/debug" "-L" "$DIR/$DIR/../../../target/$TMP/$TRIPLE/debug" "--out-dir" "$TMP "tests/actual_tests_bless/wrong_diagnostic_code.rs" "--edition" "2021"
+
+error: diagnostic code `should_be_dead_code` not found on line 3
+ --> tests/actual_tests_bless/wrong_diagnostic_code.rs:4:10
+  |
+4 |     //~^ should_be_dead_code
+  |          ^^^^^^^^^^^^^^^^^^^ expected because of this pattern
+  |
+
+error: there were 1 unmatched diagnostics
+ --> tests/actual_tests_bless/wrong_diagnostic_code.rs:3:4
+  |
+3 | fn foo() -> i32 {
+  |    ^^^ Error: function `foo` is never used
+  |
```

Comment:
Print the diagnostic code of unmatched diagnostics, otherwise it is very hard to figure out how to resolve this.

Reply:
It's been added only for errors. Should be added for warnings as well?

Reply:
the `//~ CODE` scheme does not work for warnings, so no.